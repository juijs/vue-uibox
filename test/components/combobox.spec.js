import { mount } from 'vue-test-utils'
import UixComboBox from '@/components/combobox'

describe('combobox.js', () => {
    const wrapper = mount(UixComboBox, {
        propsData: {
            items: [
                { text: 'a', value: 1 }
            ]
        }
    });

    it('renders li for each item in props.items', () => {
        expect(wrapper.vm.items.length).toBe(1);
    })

    it('server render', () => {
        wrapper.vm.$mount();

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
})