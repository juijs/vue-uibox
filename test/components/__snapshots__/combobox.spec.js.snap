// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`combobox.js server render 1`] = `
<div
  class="combo"
  value="1"
>
  <a
    class="btn small"
  >
    a
  </a>
  <ul
    style="overflow-x: hidden; overflow-y: auto; max-height: 100px;"
  >
    <li>
      <a
        href="#"
        value="1"
      >
        a
      </a>
    </li>
  </ul>
   
  <a
    class="btn toggle small"
  >
    <i
      class="icon-arrow2"
    />
  </a>
   
</div>
`;
